app/controllers/application_controller.rb

```diff
@@ -1,2 +1,15 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```

Comment:
Authentication filter should be added to only those controllers in which we want user to be signed in. Adding it in application controller will make the whole application closed to signed in users only.

Reply:
a user always need to signed in e.g, to like post or comment on a post or send a friend request etc.  that's why i put it in application controller.